web/src/App.res

```diff
@@ -69,6 +69,51 @@ module MonocleNav = {
   }
 }
 
+module Login = {
+  module Modal = {
+    @react.component
+    let make = (~store: Store.t, ~setShowLoginModal) => {
+      let loginTitle = "Set your username"
+      let (_, dispatch) = store
+      let (username, setUsername) = React.useState(_ => "")
+      let onChange = (value, _) => setUsername(_ => value)
+      let onClick = _ => {
+        username->Login->dispatch
+        setShowLoginModal(_ => false)
+      }
+      let isDisabled = username == ""
+
+      <LoginPage loginTitle>
+        <Form>
+          <FormGroup label={"Username"} fieldId={"login"}>
+            <TextInput id={"login"} onChange value={username} />
+          </FormGroup>
+          <ActionGroup>
+            <Button _type=#Submit variant=#Primary onClick isDisabled> {"Login"} </Button>
+          </ActionGroup>
+        </Form>
+      </LoginPage>
+    }
+  }
+  module Button = {
+    @react.component
+    let make = (~store: Store.t, ~setShowLoginModal) => {
+      let (state, dispatch) = store
+      let onClickLogin = _ => setShowLoginModal(_ => true)
+      let onClickLogout = _ => Logout->dispatch
+      <div style={ReactDOM.Style.make(~paddingRight="13px", ())}>
+        {switch state.username {
+        | Some(username) =>
+          <Button variant=#Tertiary icon={<Patternfly.Icons.User />} onClick={onClickLogout}>
```

Comment:
Perhaps this could use a tooltip `Click here to logout` ?
Or the otherway around, the button could show `Logout` with a tooltip that indicate the current username?